commcare_connect/users/views.py

```diff
@@ -119,22 +120,44 @@ def start_learn_app(request):
     return HttpResponse(status=200)
 
 
-@require_GET
-def accept_invite(request, invite_id):
-    try:
-        o = OpportunityAccess.objects.get(invite_id=invite_id)
-    except OpportunityAccess.DoesNotExist:
-        return HttpResponse("This link is invalid. Please try again", status=404)
-    with transaction.atomic():
-        o.accepted = True
-        o.save()
-        user_invite = UserInvite.objects.get(opportunity_access=o)
-        user_invite.status = UserInviteStatus.accepted
-        user_invite.save()
-    return HttpResponse(
-        "Thank you for accepting the invitation. Open your CommCare Connect App to "
-        "see more information about the opportunity and begin learning"
-    )
+class AcceptInviteView(View):
+    def get(self, request, invite_id):
+        try:
+            access = OpportunityAccess.objects.get(invite_id=invite_id)
+        except OpportunityAccess.DoesNotExist:
+            return HttpResponse("This link is invalid. Please try again", status=404)
+        get_token(request)
+        context = {
+            "opportunity_name": access.opportunity.name,
+        }
+        return render(request, "users/accept_invite_confirm.html", context=context)
```

Comment:
nit: the context can be directly passed as a dict here directly instead of declaring a variable and then passing it.